### ISSUE TITLE
TouchableOpacity responds to wrong area when wrapping an svg

### ISSUE THREAD

**Issue by perqin** [facebook/react-native#35675]
### Description

We encounter this issue on upgrade React Native from 0.64.4 to 0.68.2. After creating minimal reproducible demo, we find that this also happens on latest version (0.70.6), but it works on 0.64.4.

### Version

0.70.6

### Output of `npx react-native info`

```
info Fetching system and libraries information...
System:
    OS: macOS 13.1
    CPU: (8) x64 Apple M2
    Memory: 47.66 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.20.0 - ~/.nvm/versions/node/v14.20.0/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v14.20.0/bin/yarn
    npm: 6.14.17 - ~/.nvm/versions/node/v14.20.0/bin/npm
    Watchman: 2022.10.10.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK:
      API Levels: 28, 29, 30, 31, 32, 33
      Build Tools: 29.0.2, 30.0.2, 30.0.3, 33.0.0
      System Images: android-30 | ARM 64 v8a, android-33 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: Dolphin 2021.3.1 Patch 1 Dolphin 2021.3.1 Patch 1
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.16.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0
    react-native: 0.70.6 => 0.70.6
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

1. Install and configure `react-native-svg`;
2. Wrap an svg icon with `TouchableOpacity`;
3. Click outside the icon.

Here's the minimal reproducible code:

```tsx
import React from 'react';
import {
  SafeAreaView,
  StatusBar,
  Text,
  TouchableOpacity,
  useColorScheme,
  View,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';
import Svg from 'react-native-svg';
import IcPin from './ic-pin-grey-2.svg';

const App = () => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <TouchableOpacity
        style={[{backgroundColor: 'green', marginTop: 200}]}
        onPress={() => {
          console.info('press pin');
        }}>
        <View>
          <Svg width={18} height={23}>
            <IcPin />
          </Svg>
        </View>
        <Text>HELLO</Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
};

export default App;
```

#### Actual behavior

The click is responded by the `TouchableOpacity`. Touch effect shows and the `onPress` callback is called. See the video below:

https://user-images.githubusercontent.com/7088631/208367766-f13c1fe9-c5a8-4008-81a3-14bc7464918f.mp4

#### Expected behavior

The click should only be responded when clicking inside the rendering area of the view wrapped by `TouchableOpacity`.

### Snack, code example, screenshot, or link to a repository

https://github.com/perqin/react-native-touchable-svg-issue

The repo root has 2 directory, `SvgIssue0644` which uses React Native 0.64.4 and `SvgIssue0706` which uses 0.70.6.

**Comment by perqin** [comment]
Any updates?

**Comment by alymenbr** [comment]
I was having the same problem on RN 0.71.12. In my case, I was using png's instead of svg's. 

What I had to do is take the image out of the touchable, then absolute-position it so it ends up in the same place as when it was contained by the original touchable. To keep the touch behaviour of the image, I added a touchable containing it with no margins or paddings.

ORIGINAL:
```
    <View>
      <TouchableHighlight style={{width: '100%'}}>
        <Image />
      </TouchableHighlight>
    </View>
```

MODIFIED ONE:
```
    <View>
      <TouchableHighlight style={{width: '100%'}} />
      <TouchableHighlight style={{position: 'absolute', top: 0}}>
        <Image />
      </TouchableHighlight>      
    </View>
```

This may seem the same, but it isn't:
 - The original had a touchable with an image inside spanning the entire width of the container.
 - The new one has a touchable with NO image inside spanning the entire width of the container. 
 -  The new one has an additional touchable containing the image, fitting exactly the image dimensions (no padding or margins added).
